analytics/terraform/spark-eks-ipv6/vpc.tf

```diff
@@ -0,0 +1,36 @@
+# VPC with IPv6 capability
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 5.0"
+
+  name = local.name
+  cidr = local.vpc_cidr
+
+  azs             = local.azs
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 4, k)]
+  public_subnets  = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k + 48)]
+
+  enable_nat_gateway     = true
+  single_nat_gateway     = true
+  enable_ipv6            = true
```

Comment:
with IPv6 clusters we'll have to use Prefix Delegation, which we should set up CIDR reservations for https://aws.github.io/aws-eks-best-practices/networking/prefix-mode/index_linux/#use-subnet-reservations-to-avoid-subnet-fragmentation-ipv4 

For this solution I think we're assuming that the EKS environment is the only thing running in the VPC so the fragmentation isn't really an issue. This may be worth calling out in the docs as something the customer should add (or we can add it in as another PR later)?

Reply:
Added to vpc cni
```
configuration_values = jsonencode({
  env = {
    ENABLE_PREFIX_DELEGATION = "true"
    WARM_PREFIX_TARGET       = "1"
  }
})
```